website/docs/tutorials/08-cairo-1-support/04-compiling.md

```diff
@@ -0,0 +1,62 @@
+---
+sidebar_label: Compilation
+---
+
+# Compiling a Cairo 1 Project with Protostar
+
+To compile your Starknet contracts written in Cairo 1 using Protostar, follow these steps:
+
+1. [Prepare your contracts properly](./03-contracts.md#prepare-contracts).
+2. Specify the contracts and their corresponding Cairo source files in the [`protostar.toml` configuration file](/docs/tutorials/configuration-file).
+Each Cairo file that contains an entrypoint function should be listed in the [`contracts` section of the configuration file](/docs/tutorials/configuration-file#contracts).
+An entrypoint is a function decorated with `#[external]` or `#[view]`.
+3. Run the [`protostar build-cairo1`](../../cli-reference.md#build-cairo1) command.
+
+## Output directory
+
+Running [`protostar build-cairo1`](../../cli-reference.md#build-cairo1) command will generate the compiled contracts in the `build` directory by default.
+You can specify a custom output directory using the [`--compiled-contracts-dir`](../../cli-reference#compiled-contracts-dir-pathbuild-1) argument.
+
+```
+$ protostar build-cairo1 --compiled-contracts-dir out
+```
+
+This will create the following files in the `out` directory:
+
+```
+protostar-project
+├── src
+│   ├── main.cairo
+│   └── proxy.cairo
+├── out
+│   ├── main.json
+│   ├── proxy.json
+└── protostar.toml
+```
+
+## External libraries
+
+If you want to attach external libraries to the build, you can specify additional paths that will be searched for the cairo libraries. To do that, you need to use the [`--cairo-path`](../../cli-reference.md#cairo-path-path-1) argument like this:
```

Comment:
We should note that this needs to be a library

Reply:
[address feedback](https://github.com/software-mansion/protostar/pull/1643/commits/2b2442e737d874a66c2a786391e54e7c673e005b)